File: lib/src/main/java/net/mm2d/upnp/ControlPoints.java
/*
 * Copyright (c) 2018 大前良介 (OHMAE Ryosuke)
 *
 * This software is released under the MIT License.
 * http://opensource.org/licenses/MIT
 */

package net.mm2d.upnp;

import net.mm2d.upnp.empty.EmptyAction;
import net.mm2d.upnp.empty.EmptyControlPoint;
import net.mm2d.upnp.empty.EmptyDevice;
import net.mm2d.upnp.empty.EmptyService;
import net.mm2d.upnp.empty.EmptySsdpMessage;

public final class ControlPoints {
    private ControlPoints() {
    }

    private static final ControlPoint EMPTY_CONTROL_POINT = new EmptyControlPoint();

    public static ControlPoint emptyControlPoint() {
        return EMPTY_CONTROL_POINT;
    }

    private static final Device EMPTY_DEVICE = new EmptyDevice();

    public static Device emptyDevice() {
        return EMPTY_DEVICE;
    }

    private static final Service EMPTY_SERVICE = new EmptyService();

    public static Service emptyService() {
        return EMPTY_SERVICE;
    }

    private static final Action EMPTY_ACTION = new EmptyAction();

    public static Action emptyAction() {
        return EMPTY_ACTION;
    }

    private static final SsdpMessage EMPTY_SSDP_MESSAGE = new EmptySsdpMessage();

    public static SsdpMessage emptySsdpMessage() {
        return EMPTY_SSDP_MESSAGE;
    }
}

File: mmupnp/src/main/java/net/mm2d/upnp/cp/ControlPoints.kt
/*
 * Copyright (c) 2018 大前良介 (OHMAE Ryosuke)
 *
 * This software is released under the MIT License.
 * http://opensource.org/licenses/MIT
 */

package net.mm2d.upnp.cp

import net.mm2d.upnp.common.SsdpMessage
import net.mm2d.upnp.cp.empty.*

/**
 * Provide EmptyObject.
 *
 * @author [大前良介 (OHMAE Ryosuke)](mailto:ryo@mm2d.net)
 */
object ControlPoints {
    /**
     * Returns the ControlPoint's Empty Object.
     *
     * @return ControlPoint's Empty Object
     */
    fun emptyControlPoint(): ControlPoint = EmptyControlPoint

    /**
     * Returns the Device's Empty Object.
     *
     * @return Device's Empty Object
     */
    fun emptyDevice(): Device = EmptyDevice

    /**
     * Returns the Service's Empty Object.
     *
     * @return Service's Empty Object
     */
    fun emptyService(): Service = EmptyService

    /**
     * Returns the Action's Empty Object.
     *
     * @return Action's Empty Object
     */
    fun emptyAction(): Action = EmptyAction

    /**
     * Returns the SsdpMessage's Empty Object.
     *
     * @return SsdpMessage's Empty Object
     */
    fun emptySsdpMessage(): SsdpMessage = EmptySsdpMessage
}
